azurerm_windows_function_app requires both connection string and instrumentation key for app insights

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

0.14.11

### AzureRM Provider Version

3.0.2

### Affected Resource(s)/Data Source(s)

azurerm_windows_function_app

### Terraform Configuration Files

```hcl
-
```


### Debug Output/Panic Output

```shell
Error: Missing required argument

  on cdmutil.tf line 40, in resource "azurerm_windows_function_app" "cdmutil_fa":
  40:     application_insights_key = azurerm_application_insights.cdmutil_ai.instrumentation_key

"site_config.0.application_insights_key": all of
`site_config.0.application_insights_connection_string,site_config.0.application_insights_key`
must be specified

##[error]Error: Terraform Plan failed with exit code: 1
```


### Expected Behaviour

Only one of the settings should be required not both

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

Same issue as with the linux equiv that was fixed https://github.com/hashicorp/terraform-provider-azurerm/issues/16077

--- a/internal/services/appservice/helpers/function_app_slot_schema.go
+++ b/internal/services/appservice/helpers/function_app_slot_schema.go
@@ -94,23 +94,17 @@ func SiteConfigSchemaWindowsFunctionAppSlot() *pluginsdk.Schema {
 				},
 
 				"application_insights_key": {
-					Type:      pluginsdk.TypeString,
-					Optional:  true,
-					Sensitive: true,
-					RequiredWith: []string{
-						"site_config.0.application_insights_connection_string",
-					},
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					Sensitive:    true,
 					ValidateFunc: validation.StringIsNotEmpty,
 					Description:  "The Instrumentation Key for connecting the Windows Function App to Application Insights.",
 				},
 
 				"application_insights_connection_string": {
-					Type:      pluginsdk.TypeString,
-					Optional:  true,
-					Sensitive: true,
-					RequiredWith: []string{
-						"site_config.0.application_insights_key",
-					},
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					Sensitive:    true,
 					ValidateFunc: validation.StringIsNotEmpty,
 					Description:  "The Connection String for linking the Windows Function App to Application Insights.",
 				},

